### ISSUE TITLE
If merging a class and namespace, the class can't reference types in the namespace

### ISSUE THREAD

**Issue by tomdale** [Microsoft/TypeScript#10544]
**TypeScript Version:**  nightly (Version 2.1.0-dev.20160825)

**Code**

``` ts
class Logger {
  public logLevel = Logger.LogLevel.Info;
}

namespace Logger {
  export enum LogLevel {
    Verbose,
    Info
  }
}

export default Logger;
```

Note that this error only seems to occur when `--declaration` is passed to `tsc`.

**Expected behavior:**

TypeScript should treat the merged class and namespace as a single entity.

**Actual behavior:** 

TypeScript complains that the class references a type in the namespace:

> error TS4031: Public property 'logLevel' of exported class has or is using private name 'default.LogLevel'.

**Comment by mhegazy** [comment]
What is the file and the position this is reported on. i can not reporoduce this locally.

```cmd
c:\ts>type c:\test\sandbox6\b.d.ts
declare class Logger {
    private logLevel;
}
declare namespace Logger {
    enum LogLevel {
        Verbose = 0,
        Info = 1,
    }
}
export default Logger;

c:\ts>tsc --v
Version 2.1.0-dev.20160825

c:\ts>type c:\test\sandbox6\b.ts
class Logger {
  private logLevel = Logger.LogLevel.Info;
}

namespace Logger {
  export enum LogLevel {
    Verbose,
    Info
  }
}

export default Logger;
c:\ts>tsc --v
Version 2.1.0-dev.20160825

c:\ts>tsc c:\test\sandbox6\b.ts --d

c:\ts>echo %ERRORLEVEL%
0

```

**Comment by tomdale** [comment]
@mhegazy Please see https://github.com/tomdale/typescript-repro/tree/10544.

```sh
merged-class-namespace :: (10544) » cat src/logger.ts
class Logger {
  public logLevel = Logger.LogLevel.Info;
}

namespace Logger {
  export enum LogLevel {
    Verbose,
    Info
  }
}

export default Logger;


merged-class-namespace :: (10544) » cat tsconfig.json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,

    "moduleResolution": "node",
    "target": "es2015",
    "module": "commonjs",
    "inlineSourceMap": true,
    "declaration": true,

    "rootDir": "src",
    "outDir": "dist"
  },

  "include": [
    "src/**/*"
  ]
}


merged-class-namespace :: (10544) » tsc
src/logger.ts(2,3): error TS4031: Public property 'logLevel' of exported class has or is using private name 'default.LogLevel'.
```

**Comment by tomdale** [comment]
Also, if there is a workaround to this, I'd love to know—it's blockingish me on some stuff I'm working on.

**Comment by DanielRosenwasser** [comment]
Try using an export clause instead

```ts
class Logger {
  public logLevel = Logger.LogLevel.Info;
}

namespace Logger {
  export enum LogLevel {
    Verbose,
    Info
  }
}

export { Logger as default };
```

**Comment by mhegazy** [comment]
i see it now. thanks for the repro. i was using a private which does not trigger the issue.

**Comment by tomdale** [comment]
@DanielRosenwasser Perfect, thank you!
@mhegazy I did the same thing when making the reduced case initially. :)

**Comment by DanielRosenwasser** [comment]
@mhegazy with my suggested fix on that repro, I see a bug in the LS as soon as I try to consume it. I'll try to create a separate issue.

**Comment by DanielRosenwasser** [comment]
Filed #10546. @tomdale you shouldn't be affected much by that.

**Comment by RyanCavanaugh** [comment]
Seems to have been fixed
